test/hummingbot/connector/exchange/latoken/test_latoken_api_order_book_data_source.py

```diff
@@ -0,0 +1,548 @@
+import asyncio
+import json
+import re
+import unittest
+from decimal import Decimal
+from test.hummingbot.connector.network_mocking_assistant import NetworkMockingAssistant
+from typing import Any, Awaitable, Dict, List
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from aioresponses.core import aioresponses
+from bidict import bidict
+
+import hummingbot.connector.exchange.latoken.latoken_constants as CONSTANTS
+import hummingbot.connector.exchange.latoken.latoken_web_utils as web_utils
+from hummingbot.connector.exchange.latoken.latoken_api_order_book_data_source import LatokenAPIOrderBookDataSource
+from hummingbot.connector.time_synchronizer import TimeSynchronizer
+from hummingbot.core.api_throttler.async_throttler import AsyncThrottler
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+
+
+class LatokenAPIOrderBookDataSourceUnitTests(unittest.TestCase):
+    # logging.Level required to receive logs from the data source logger
+    level = 0
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.ev_loop = asyncio.get_event_loop()
+        cls.base_asset = "d8ae67f2-f954-4014-98c8-64b1ac334c64"
+        cls.quote_asset = "0c3a106d-bde3-4c13-a26e-3fd2394529e5"
+        cls.trading_pair = "ETH-USDT"
+        # cls.ex_trading_pair = cls.base_asset + cls.quote_asset
+        cls.domain = "com"
+        for task in asyncio.all_tasks(loop=cls.ev_loop):
+            task.cancel()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        for task in asyncio.all_tasks(loop=cls.ev_loop):
+            task.cancel()
```

Comment:
we put super in the tearDownClass, not sure why?

Reply:
It is required because `tearDownClass` is a method defined by the unit test framework. It has logic in the superclass, and if super() is not called then the logic defined in the superclass is not executed.